docs/deployment/pull-backup.rst

```diff
@@ -84,27 +72,26 @@ already installed on the client.
     # Mount BorgBackup repository inside it.
     mkdir /tmp/sshfs/borgrepo
     mount --bind /path/to/repo /tmp/sshfs/borgrepo
-    # Make borg executable available.
-    cp /usr/local/bin/borg /tmp/sshfs/usr/local/bin/borg
+    # Make borg executable available as /pull-borg.
+    cp /usr/local/bin/borg /tmp/sshfs/pull-borg
     # Mount important system directories and enter chroot.
     cd /tmp/sshfs
     for i in dev proc sys; do mount --bind /$i $i; done
     chroot /tmp/sshfs
 
 Now we are on the backup system but inside a chroot with the client's root file
-system. We have a copy of Borg binary in ``/usr/local/bin`` and the repository
-in ``/borgrepo``. Borg will back up the client's user/group names, and we can
+system. We have a copy of Borg binary as ``/pull-borg`` and the repository in
+``/borgrepo``. Borg will back up the client's user/group names, and we can
 create the backup, retaining the original paths, excluding the repository:
 
 ::
 
-    borg create --exclude /borgrepo --files-cache ctime,size /borgrepo::archive /
+    /pull-borg create --exclude /borgrepo --files-cache ctime,size /borgrepo::archive /
 
 For the sake of simplicity only ``/borgrepo`` is excluded here. You may want to
-set up an exclude file with additional files and folders to be excluded. Also
-note that we have to modify Borg's file change detection behaviour – SSHFS
-cannot guarantee stable inode numbers, so we have to supply the
-``--files-cache`` option.
+set up an exclude file with additional files and folders to be excluded. You
+should not execute any other command besides ``/pull-borg`` if you do not trust
+the client file system!
```

Comment:
The scenario states that we are in control of client and server, and our concern in this scenario is integrity, not security. Of course we could add more security by more complexity. I want to keep this scenario as simple as possible, but we could add some lines to show what to consider for more security. Or we add another scenario that adds the readonly fs you propose.

One could also discuss if the BorgBackup binary should be installed by the server or by the client. Here, the client trusts the server binary, but this could also be inverted so that the server expects/trusts a BorgBackup binary on the client.

Reply:
Well, this doesn't look that complex, can you try it?:
```
mount --bind -o ro /tmp/borg_bin /chroot/borg_bin
```

The main reason a lot of people want to do pull backups is that they fear an attacker gets control of the borg client system. If the documented pull mode does not solve this, it gets a bit pointless for many.